paper/paper.md

```diff
@@ -0,0 +1,167 @@
+likely a secondary interest—teaching students through traditional computer
+science offerings is not always effective [@Cortina2007; @Forte2005;
+@Guzdial2005]. While there are visual programming languages such as Raptor
+[@Carlisle2004] and Scratch [@Resnick2009] that are intended to be easy and
+enjoyable for those unfamiliar with programming to learn, they lack specialized
+tools such as numerical or data visualization libraries, which are required for
+effective work in domain sciences. A hybrid approach is to teach students
+computational concepts in an interactive environment where it is possible to
+quickly write, test, and tweak small units of code. Many such environments
+exist, including Mathematica [@mathematica], Maple [@maple], MATLAB [@matlab],
+Sage [@sage] and IPython [@PerezGranger2007].
+
+In recent years, the IPython project introduced the *Jupyter notebook*
+[@kluyver2016jupyter], an interface that is conducive to interactive and
+literate computing, where programmers can interleave prose with code and
+figures. The Jupyter notebook is ideal for educators because it allows them to
+create assignments which include instructions along with code or Markdown cells,
+in which students can provide solutions to exercises. Students can, for example,
+write code both to compute and visualize a particular result. Because the
+notebook is interactive, students can iterate on a coding problem without
+needing to switch back and forth between a command line and a text editor, and
+they can rapidly see results alongside the code which produced them.
+
+Instructors in many fields have begun using the Jupyter notebook as a teaching
+platform. The notebook has appeared in over 70 classes
+[@Castano_Jupyter_Map_Dataset] on subjects including geology, mathematics,
+mechanical engineering, data science, chemical engineering, and bioinformatics.
+Software Carpentry, an organization that aims to teach graduate students basic
+computational skills, has also adopted the notebook for some of its lessons
+[@Wilson2014].
+
+Despite its appearance in many classrooms, before to the existence of nbgrader,
+the notebook was rarely used on a large scale for *graded* assignments. Instead,
+it was often used either for ungraded in-class exercises, or in classes small
+enough that notebooks could be graded by hand. This is because there are several
+challenges to using the notebook for graded assignments at scale. First, for
+large classes, it is not feasible for an instructor to manually grade the code
+that students write: there must be a way to autograde the assignments. However,
+a Jupyter notebook is not a typical script that can be run and may contain
+multiple parts of a problem within the same notebook; thus, autograding a
+notebook is less straightforward than it is for a traditional script. Second,
+for many courses, programming is a means to an end: understanding concepts in a
```

Comment:
Even in CS, programming is not the goal. Isn't it always about the concepts?

Reply:
In all but 1 or 2 CS classes that I've taken, the assignments did not involve any interpretation of the results of the code at all. If there were free responses they were for separate things, like deriving equations. I think this is pretty typical of CS classes, but I could be wrong. 
Regardless though I don't think there's anything in this paragraph that implies this second point only applies to fields outside CS?

Reply:
Ok, sounds good.